inc/hclib.h

```diff
@@ -174,6 +175,14 @@ void hclib_end_finish();
 hclib_future_t *hclib_end_finish_nonblocking();
 void hclib_end_finish_nonblocking_helper(hclib_promise_t *event);
 
+
+#define _HCLIB_FINISH_IMPL(END_FUNC) \
```

Comment:
I used this macro in at least one of the tests. It's nice to have in C code since it makes finish scopes more obvious and helps ensure that the start/end calls always match. It's also nice in C++ since it lets you avoid unnecessarily introducing another lambda with another capture list vs the `hclib::finish` function.